cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -613,6 +618,14 @@ private void deleteApp(Id.Application appId, ApplicationSpecification spec) thro
     }
   }
 
+  // TODO: CDAP-5427 - This should be a single operation
+  private void revokePrivileges(ApplicationId appId, ApplicationSpecification appSpec) throws Exception {
+    authorizerInstantiatorService.get().revoke(appId);
```

Comment:
Does not, and like the comment says, this should be a single call eventually. However, it does seem more logical to revoke on the programs first, then the app. Fixed.